base/src/commonMain/kotlin/state/KalugaState.kt

```diff
@@ -348,17 +351,22 @@ abstract class BaseHotStateRepo<S : KalugaState, F : MutableSharedFlow<S>>(
 
     abstract val lazyMutableSharedFlow: Lazy<F>
 
-    // guards once only initialization across threads
-    private val initialized = AtomicBoolean(false)
+    private val lock = Mutex()
+    private var initialized = false
 
     override val mutableFlow: F
         get() {
             val isInitialized = lazyMutableSharedFlow.isInitialized()
             val flow = lazyMutableSharedFlow.value
-            if (!isInitialized && initialized.compareAndSet(expected = false, new = true))
-                launch(coroutineContext) {
+            launch(coroutineContext) {
```

Comment:
atomicfu again, well you get the point, but this shows some of the issues again (it's async).